client/state/themes/selectors.js

```diff
@@ -357,7 +357,7 @@ export function getThemeDetailsUrl( state, theme, siteId ) {
  * @return {?String}        Theme setup instructions URL
  */
 export function getThemeSupportUrl( state, theme, siteId ) {
-	if ( ! theme || ! isPremiumTheme( theme ) ) {
+	if ( isJetpackSite( state, siteId ) || ! theme || ! isThemePremium( state, theme.id ) ) {
```

Comment:
wpcom user lands on Pressable after AT, he has a Premium theme that he has bought while on wpcom. That theme has support url. I think that he should still see that link but now `isJetpackSite( state, siteId )` blocks it.